javalib/src/main/scala/java/lang/Byte.scala

```diff
@@ -2,43 +2,55 @@ package java.lang
 
 import scalanative.runtime.{byteToUInt, byteToULong}
 
-final class Byte(override val byteValue: scala.Byte)
-    extends Number
-    with Comparable[Byte] {
+final class Byte(val _value: scala.Byte) extends Number with Comparable[Byte] {
   @inline def this(s: String) =
     this(Byte.parseByte(s))
 
+  @inline override def byteValue(): scala.Byte =
+    _value
+
   @inline override def shortValue(): scala.Short =
-    byteValue.toShort
+    _value.toShort
 
-  @inline def intValue(): scala.Int =
-    byteValue.toInt
+  @inline override def intValue(): scala.Int =
+    _value.toInt
 
-  @inline def longValue(): scala.Long =
-    byteValue.toLong
+  @inline override def longValue(): scala.Long =
+    _value.toLong
 
-  @inline def floatValue(): scala.Float =
-    byteValue.toFloat
+  @inline override def floatValue(): scala.Float =
+    _value.toFloat
 
-  @inline def doubleValue(): scala.Double =
-    byteValue.toDouble
+  @inline override def doubleValue(): scala.Double =
+    _value.toDouble
 
   @inline override def equals(that: Any): scala.Boolean =
     that match {
-      case that: Byte =>
-        byteValue == that.byteValue
+      case that: java.lang.Byte =>
+        _value == that._value
       case _ =>
         false
     }
 
   @inline override def hashCode(): Int =
-    Byte.hashCode(byteValue)
+    _value
 
   @inline override def compareTo(that: Byte): Int =
-    Byte.compare(byteValue, that.byteValue)
+    Byte.compare(_value, that._value)
 
   @inline override def toString(): String =
-    Byte.toString(byteValue)
+    Byte.toString(_value)
+
+  @inline override def __scala_==(other: _Object): scala.Boolean =
+    other match {
+      case other: java.lang.Byte    => _value == other._value
+      case other: java.lang.Short   => _value.toShort == other._value
+      case other: java.lang.Integer => _value.toInt == other._value
+      case other: java.lang.Long    => _value.toLong == other._value
+      case other: java.lang.Float   => _value.toFloat == other._value
+      case other: java.lang.Double  => _value.toDouble == other._value
+      case _                        => super.__scala_==(other)
```

Comment:
Also, I suggest just writing `_value == other._value` in all the cases. Let the primitive `==` (on primitive operands) do the appropriate conversions in all cases. I think it would be safer.